t/expansion/noexpand_conditionals.tex

```diff
@@ -0,0 +1,31 @@
+% ifx test with noexpand
+\def\foo{foo}%
+\def\bar{bar}%
+\expandafter\expandafter%
+    \expandafter\ifx%
+        \expandafter\noexpand\expandafter\foo\noexpand\bar yes%
+    \else no%
+    \fi%
+
+\def\foo{foo}%
+\expandafter\ifx\noexpand\foo\relax yes\else no\fi%
+
+% ifcat tests
+\catcode`[=13 \catcode`]=13 \def[{*}
+ifcat 1: \ifcat\noexpand[\noexpand] true \else \false\fi
+
+ifcat 2: \ifcat[* true \else false \fi
+
+ifcat 3: \ifcat\noexpand[* true \else false\fi
+
+% Advanced from xparse.sty: \ifx on "\noexpand\count" and "\count" is true, as counts are not expandable
+\def\itoii#1#2#3{#1#2}
+\chardef\eend=0\relax
+\newcount\index
+\index=4
+
+\expandafter\ifx\noexpand\index\index\else\expandafter\itoii\fi
+\expandafter\eend\the\index
```

Comment:
Ideally this test would be smaller and less confusing, but I am unsure how to reduce it without losing the bug in master (since the actual logic is pretty convoluted, and it might just be the case that you need all pieces to get a visible difference).